addon/lib/Feature.jsm

```diff
@@ -8,53 +8,108 @@ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
-const EXPORTED_SYMBOLS = this.EXPORTED_SYMBOLS = ["Feature"];
+const EXPORTED_SYMBOLS = this.EXPORTED_SYMBOLS = ["feature"];
 
+const BASERESOURCE = "perception-study";
 
 XPCOMUtils.defineLazyModuleGetter(this, "RecentWindow",
   "resource:///modules/RecentWindow.jsm");
-
+XPCOMUtils.defineLazyModuleGetter(this, "config",
+  `resource://${BASERESOURCE}/Config.jsm`);
+XPCOMUtils.defineLazyModuleGetter(this, "studyUtils",
+  `resource://${BASERESOURCE}/StudyUtils.jsm`);
 
 class Feature {
-  startup(config) {
-    console.log(`Starting the thing with ${config}`);
-    return "started";
+  constructor() {
+    this.notificationBar = new Notification();
+    this.TREATMENTS = {
+      control() {
+        // TODO glind: add control treatment
+      },
+      notification() {
+        const recentWindow = this.getMostRecentBrowserWindow();
+        if (recentWindow && recentWindow.gBrowser) {
+          this.notificationBar.show(recentWindow.document, () => {
+            // TODO bdanforth: add callback for what to do if user clicks on 'accept button'
+            console.log("Doing a thing after button accept click!");
+            // TODO glind: send correct telemetry when user clicks on button
+            studyUtils.telemetry({ event: "engagedPrompt" });
+          }, () => {
+            // TODO bdanforth: add callback for what to do if user clicks on 'reject button'
+            console.log("Doing a thing after button reject click!");
+            // TODO glind: send correct telemetry when user clicks on button
+            studyUtils.telemetry({ event: "engagedPrompt" });
+          });
+        }
+        // TODO glind: send correct telemetry when user is prompted with notification bar
+        studyUtils.telemetry({ event: "prompted"});
+      },
+      otherTreatment() {
+        // TODO glind: add other treatment(s) as needed.
+      },
+    };
+  }
+
+  startup(variation) {
+    if (variation in this.TREATMENTS) {
+      this.TREATMENTS[variation].call(this);
+      return "started";
+    }
+    return `Error: ${variation} not a valid treatment name.`;
   }
-  shutdown () {
-    console.log('shutting down feature');
+
+  shutdown() {
+    console.log("shutting down feature");
     return "shutdown";
+  }
 
+  getMostRecentBrowserWindow() {
+    return RecentWindow.getMostRecentBrowserWindow({
+      private: false,
+      allowPopups: false,
+    });
   }
 }
 
 class Notification {
   // functions related to notification box
-  constructor(props) {
+  constructor() {
     // state
     this.notice = null;
     this.notificationBox = null;
   }
-  show(doc, onClickButtonCallback) {
-    let { notice, notificationBox} = this
+  show(doc, onClickButtonAcceptCallback, onClickButtonRejectCallback) {
 
     // only one at a time is allowed
-    if (notice && notificationBox) {
-      notificationBox.removeNotification(notice);
+    if (this.notice && this.notificationBox) {
```

Comment:
I think changing these all back to "this" accessors is worse, and harder to read everywhere